### PR TITLE
Add edit access code route to organizer's dashboard

### LINKED ISSUE #1362: Make action buttons in access code list functional
**Is your feature request related to a problem? Please describe.**
Currently action buttons in access code list are not functional.

**Describe the solution you'd like**
Add necessary actions to make them functional

### PR DESCRIPTION
**Pull request by pradeepgangwar**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Add edit route to access codes, thus allowing organizer to edit a access code.

#### Changes proposed in this pull request:
- Add necessary routes and actions to edit a access code.

Fixes #1362

### CODE REVIEW

File: app/components/forms/events/view/edit-access-code.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      }
+    };
+  },
+  accessCode : '',
+  accessLink : computed('data.code', function() {
+    const params = this.get('router._router.currentState.routerJsState.params');
+    const origin = this.get('fastboot.isFastBoot') ? `${this.get('fastboot.request.protocol')}//${this.get('fastboot.request.host')}` : location.origin;
+    let link = origin + this.get('router').urlFor('public', params['events.view'].event_id, { queryParams: { access_code: this.get('data.code') } });
+    this.set('data.accessUrl', link);
+    return link;
```
Comment (srv-twry) [inline app/components/forms/events/view/edit-access-code.js]: Why this in edit ? We will have it in the access code right ?

File: app/components/forms/events/view/edit-access-code.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        this.set('allTicketTypesChecked', false);
+      }
+    },
+    submit(data) {
+      this.onValid(() => {
+        this.sendAction('save', data);
+      });
+    }
+  }
+
```
Comment (srv-twry) [inline app/components/forms/events/view/edit-access-code.js]: extra space.

File: app/components/forms/events/view/edit-access-code.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        accessCode: {
+          identifier : 'access_code',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.get('l10n').t('Please enter access code')
+            },
+            {
+              type  : 'regExp',
+              value : '^[a-zA-Z0-9_-]*$'
```
Comment (ritikamotwani) [inline app/components/forms/events/view/edit-access-code.js]: @pradeepgangwar but all the regular expressions are in utils/validators.js file and then imported.

**Comment by pradeepgangwar** [comment]
@srv-twry I modified create-access-code template to be reused for edit route :smile: .
@ritikamotwani Files are changed now :smile: .
Please review.

**Pull request merged**